File: 2-Basic-Mathematical-Knowledge-for-Algorithms/008-Brute_Force_1/java/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        // 1. 入力処理
        int n = sc.nextInt();
        int s = sc.nextInt();

        // 2. 最適化された計算
        int count = 0;
        for (int r = 1; r <= n; r++) {
            int maxB = Math.min(n, s - r);
            count += Math.max(0, maxB);
        }

        // 3. 出力処理
        System.out.println(count);
        
        sc.close();
    }
}

File: 2-Basic-Mathematical-Knowledge-for-Algorithms/008-Brute_Force_1/go/main.go
package main

import "fmt"

func min(a, b int) int {
    if a < b {
        return a
    }
    return b
}

func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}

func main() {
    // 1. 入力処理
    var n, s int
    fmt.Scan(&n, &s)

    // 2. 最適化された計算
    count := 0
    for r := 1; r <= n; r++ {
        maxB := min(n, s-r)
        count += max(0, maxB)
    }

    // 3. 出力処理
    fmt.Println(count)
}